src/Stripe.net/Entities/BalanceTransactions/BalanceTransaction.cs

```diff
@@ -44,24 +44,14 @@ public class BalanceTransaction : StripeEntity, IHasId, IHasObject
 
         #region Expandable Source
         [JsonIgnore]
-        public string SourceId { get; set; }
+        public string SourceId => this.InternalSource.Id;
```

Comment:
The Go and Java implementations are pretty different. The approach I took here is very similar to stripe-java's, with an `ExpandableField<>` generic and manually defined accessors that map to the ID string and typed expanded object within the generic instance.

We can't use stripe-go's way because we use interfaces for polymorphic fields. When an expandable polymorphic field is unexpanded, we need a concrete type to hold the ID string. That concrete type is the `ExpandableField<>` generic. stripe-go doesn't have that issue because it uses concrete intermediary structs for polymorphic fields.

I agree that the boilerplate for defining each expandable field is annoying but unfortunately I think we'll have to live with it. This PR at least reduces the boilerplate a bit.